monetio/models/_camx_mm.py

```diff
@@ -416,23 +439,28 @@ def _calc_midlayer_height_agl(dset):
         "COL",
     ), "Check dims of z, should be [TSTEP, LAY, ROW, COL]"
 
-    mid_layer_height = np.array(dset["z"])  # height in the layer upper interface of each layer
+    if "z" in list(dset.variables):
```

Comment:
```suggestion
    if "z" in dset.variables:
```

Reply:
There are other places you could change this too. In principle, checking for membership in the dict (keys) is more efficient.